app/src/main/java/org/schabi/newpipe/local/bookmark/BookmarkFragment.java

```diff
@@ -278,19 +349,93 @@ private void showLocalDialog(final PlaylistMetadataEntry selectedItem) {
                 .show();
     }
 
-    private void showDeleteDialog(final String name, final Single<Integer> deleteReactor) {
+    private void showMergeDialog(final ArrayList<StreamEntity> streams) {
+        if (activity == null || disposables == null) {
+            return;
+        }
+
+        final DialogEditTextBinding dialogBinding =
+                DialogEditTextBinding.inflate(getLayoutInflater());
+        dialogBinding.getRoot().getContext().setTheme(ThemeHelper.getDialogTheme(requireContext()));
+        dialogBinding.dialogEditText.setHint(R.string.name);
+        dialogBinding.dialogEditText.setInputType(InputType.TYPE_CLASS_TEXT);
+
+        new AlertDialog.Builder(activity)
+                .setTitle(R.string.merge_playlists)
+                .setMessage(R.string.merge_playlist_warning)
+                .setView(dialogBinding.getRoot())
+                .setCancelable(true)
+                .setNegativeButton(R.string.cancel, null)
+                .setNeutralButton(R.string.merge_and_delete, (dialog, i) -> {
+        final String name = dialogBinding.dialogEditText.getText().toString();
+        final Toast successToast = Toast.makeText(getActivity(),
+                R.string.playlists_merged_and_delete,
+                Toast.LENGTH_SHORT);
+        localPlaylistManager.createPlaylist(name, streams)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(longs -> successToast.show());
+        localPlaylistManager.deleteMultiPlaylists(selectedLocalPlaylists)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(ignored -> { /*Do nothing on success*/ });
+        remotePlaylistManager.deleteMultiPlaylists(selectedRemotePlaylists)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(ignored -> { /*Do nothing on success*/ });
```

Comment:
You are not handling errors from these calls, and you should add them to `disposables` to make sure their lifetime is handled correctly